airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/pipeline.py

```diff
@@ -39,27 +55,105 @@ def get_test_steps(context: ConnectorContext) -> STEP_TREE:
         return []
 
 
-async def run_connector_test_pipeline(context: ConnectorContext, semaphore: anyio.Semaphore) -> ConnectorReport:
+def build_extra_params_mapping(
+    extra_params: Tuple[str], unique_steps_to_run: Dict[CONNECTOR_TEST_STEP_ID, Step]
+) -> Dict[CONNECTOR_TEST_STEP_ID, STEP_PARAMS]:
+    """Build a mapping of step id to extra params.
+    Validate the extra params and raise a ValueError if they are invalid.
+    Validation rules:
+    - The extra params must be structured as --<step_id>.<param_name>=<param_value>
+    - The step id must be one of the unique steps to run.
+
+
+    Args:
+        extra_params (Tuple[str]): The extra params provided by the user.
+        unique_steps_to_run (Dict[CONNECTOR_TEST_STEP_ID, Step]): The unique steps to run in the test pipeline.
+
+    Raises:
+        ValueError: Raised if the extra params format is invalid.
+        ValueError: Raised if the step id in the extra params is not one of the unique steps to run.
+
+    Returns:
+        Dict[CONNECTOR_TEST_STEP_ID, STEP_PARAMS]: The mapping of step id to extra params.
+    """
+    extra_params_mapping: Dict[CONNECTOR_TEST_STEP_ID, STEP_PARAMS] = {}
+    valid_step_names = [step_id.value for step_id in CONNECTOR_TEST_STEP_ID]
+    for param in extra_params:
+        is_flag = "=" not in param
+        pattern = EXTRA_PARAMS_PATTERN_FOR_FLAG if is_flag else EXTRA_PARAMS_PATTERN
+        matches = pattern.match(param)
+        if not matches:
+            raise ValueError(f"Invalid parameter {param}. {EXTRA_PARAMS_PATTERN_ERROR_MESSAGE}")
+        if is_flag:
+            step_name, param_name = matches.groups()
+            param_value = None
+        else:
+            step_name, param_name, param_value = matches.groups()
+        if step_name not in valid_step_names or CONNECTOR_TEST_STEP_ID(step_name) not in unique_steps_to_run:
+            raise ValueError(
+                f"Invalid step name {step_name}, it must be one of {[step_id.value for step_id in unique_steps_to_run.keys()]}"
+            )
+        step_id = CONNECTOR_TEST_STEP_ID(step_name)
```

Comment:
❗Given that `extra_params` is a useful concept for all pipelines we should avoid linking its parsing logic to connector tests directly.

We could either
1. Not worry if they pass a not found id (as the arg will not apply anywhere)
2. Or raise this during `run_steps` as "warning: no corresponding step found for argument in run tree"